modules/dhcp_native_ms/dhcp_native_ms_main.rb

```diff
@@ -78,6 +78,41 @@ def unused_ip(subnet_address, mac_address, from_address, to_address)
         raise Proxy::DHCP::NotImplemented.new("DhcpsApi::Server#get_free_ip_address is not available on Windows Server 2008 and earlier versions.")
       end
 
+      dhcp_range = dhcpsapi.list_subnet_elements(subnet_address, DhcpsApi::DHCP_SUBNET_ELEMENT_TYPE::DhcpIpRanges)
+      dhcp_start_addr = dhcp_range.map {|r| r[:element][:start_address]}.first
+      dhcp_end_addr = dhcp_range.map {|r| r[:element][:end_address]}.first
+
+      if from_address.to_s.empty?
+        from_address = dhcp_start_addr
+      else
+        foreman_start_addr_ip = IPAddr.new(from_address, Socket::AF_INET)
+        dhcp_start_addr_ip = IPAddr.new(dhcp_start_addr, Socket::AF_INET)
+      end
+      if to_address.to_s.empty?
+        to_address = dhcp_end_addr
+      else
+        foreman_end_addr_ip = IPAddr.new(to_address, Socket::AF_INET)
+        dhcp_end_addr_ip = IPAddr.new(dhcp_end_addr, Socket::AF_INET)
+      end
+
+      if foreman_start_addr_ip.to_i > dhcp_end_addr_ip.to_i
```

Comment:
There's a PR open (which should be merged in soon) that introduces a dedicated validator for range checks (https://github.com/theforeman/smart-proxy/pull/543/files#diff-42cb95eb87ed37db7fd2fc99a3064f06R67). I would suggest using it instead.

Reply:
There are 2 different use cases here: 
- foreman range is out of dhcp scope = warning message and exit
- foreman range is out of dhcp scope but dhcp scope is inside foreman range, in that case we just update the from/to to match the dhcp scope and we continue.
Not sure how I can use the validate_subnet_range for that cases.

Reply:
Now that I think of it, I'd say treat #2 as an error and let the user to correct the data/scope (as it may have impact on other parts of Foreman) instead of massaging it.